### ISSUE TITLE
Fish wget Alias Not Working As Expected

### ISSUE THREAD

**Issue by zQueal** [fish-shell/fish-shell#2246]
Well, not at all really.

Basically I want to create a fish function which watches for a `--clean` flag and does something special. If `--clean` is not present, then it will simply call the `wget` program binary and run like normal and pass all other file arguments onto `wget` under bash.

```
function target --description 'Hijack wget and add functionality'

switch $argv
        case --clean
                command wget --quiet $argv; and command tar xf $argv; and command rm $argv or echo 'Error!'
        case '*'
                command wget "$@" _ $argv
end

end
```

I've been looking in the documentation and can't seem to figure this out myself. My first issue is that `tar` is recognizing the url, and not the actual filename of the downloaded file. This is prompting `ssh` to step in as it recognizes it as a remote file. 

# Questions

- [ ] How do I get fish to recognize the filename of `$argv` (which is always a url) instead of the input url?
- [ ] Would this support multiple files?

**Comment by zanchey** [comment]
This may be more difficult than it appears at first. The only parameter passed to the function is the URL. The name of the downloaded file can't be reliably extracted from the URL; there may be redirects or alternative names provided, and wget tries to avoid overwriting existing files by renaming the downloaded file. [This question on StackOverflow](http://stackoverflow.com/q/8574038/125549) shows a few ways of approaching the problem; if you are going to use the `basename` method then you should definitely specify it as an argument to wget as well.

Your function as it stands at present will also behave poorly if you try and pass any options to wget. Can I ask why you want to wrap wget in this way? Why not always behave as if `--clean` was passed  if the function is called, and just use wget directly for the other cases?

**Comment by ridiculousfish** [comment]
One thought I had is to make a temporary directory, wget into that, and then use a glob to determine the resulting file.

**Comment by zQueal** [comment]
Thanks for commenting!

> Why not always behave as if --clean was passed if the function is called, and just use wget directly for the other cases?

There's a specific use case which I want to use the `--clean` flag for, so I don't really want to always remove the downloaded `.tar.gz` files. 

> One thought I had is to make a temporary directory

Would it help if I downloaded the file as a predictable temporary name, extract, then delete the temporary name that I gave it? As far as I can tell that would have the same functionality that I'm looking for. The name of the actual `tar.gz` files have little to no consequence.
